nuxeo-services/login/nuxeo-platform-login-openid/src/main/java/org/nuxeo/ecm/platform/oauth2/openid/OpenIDConnectProviderDescriptor.java

```diff
@@ -165,5 +192,42 @@ public String getUserMapper() {
         return userInfoClass;
     }
 
-
+    /**
+     * @since 11.1
+     */
+    public String getAuthenticationMethod() {
+        return authenticationMethod;
+    }
+    
+    @Override
+    public Descriptor merge(Descriptor o) {
+        OpenIDConnectProviderDescriptor other = (OpenIDConnectProviderDescriptor) o;
+        OpenIDConnectProviderDescriptor merged = new OpenIDConnectProviderDescriptor();
+        merged.name = name;
+        merged.enabled = other.enabled;
+        merged.authorizationServerURL = StringUtils.isNotBlank(other.authorizationServerURL)
+                ? other.authorizationServerURL
+                : authorizationServerURL;
+        merged.clientId = StringUtils.isNotBlank(other.clientId) ? other.clientId : clientId;
+        merged.clientSecret = StringUtils.isNotBlank(other.clientSecret) ? other.clientSecret : clientSecret;
+        merged.icon = StringUtils.isNotBlank(other.icon) ? other.icon : icon;
+        merged.scopes = ArrayUtils.isNotEmpty(other.scopes) ? other.scopes : scopes;
+        merged.tokenServerURL = StringUtils.isNotBlank(other.tokenServerURL) ? other.tokenServerURL : tokenServerURL;
+        merged.userInfoURL = StringUtils.isNotBlank(other.userInfoURL) ? other.userInfoURL : userInfoURL;
+        merged.label = StringUtils.isNotBlank(other.label) ? other.label : label;
+        merged.description = StringUtils.isNotBlank(other.description) ? other.description : description;
+        merged.accessTokenKey = !other.accessTokenKey.equals(DEFAULT_ACCESS_TOKEN_KEY) ? other.accessTokenKey
+                : accessTokenKey;
+        merged.userInfoClass = other.userInfoClass != DEFAULT_USER_INFO_CLASS ? other.userInfoClass : userInfoClass;
+        merged.redirectUriResolver = other.redirectUriResolver != DEFAULT_REDIRECT_URI_RESOLVER_CLASS
+                ? other.redirectUriResolver
+                : redirectUriResolver;
+        merged.userResolverClass = other.getUserResolverClass() != DEFAULT_USER_RESOLVER_CLASS ? other.userResolverClass
```

Comment:
Though it's not strictly needed given the implementation of `getUserResolverClass()`, it would be safer with:
```suggestion
        Class<? extends UserResolver> otherUserResolverClass = other.getUserResolverClass();
        merged.userResolverClass = otherUserResolverClass != DEFAULT_USER_RESOLVER_CLASS ? otherUserResolverClass : userResolverClass;
```